torch/_export/serialize.py

```diff
@@ -46,33 +69,411 @@ def _reverse_map(d):
+        # - This operator returns a list of Tensors.
+        # - This operator returns multiple Tensors.
+        #
+        # Either way, start by gathering a list of TensorArguments with the correct names.
+        # For consistent naming with FX, consult the downstream `getitem` node and
+        # make sure our outputs have the same name.
+        idx_to_name = {}
+        for user in node.users:
+            assert user.target is operator.getitem, f"User node {user} of {node} is incorrect"
+            idx_to_name[user.args[1]] = user.name
+
+        for idx, _ in enumerate(meta_val):
+            # FX does not emit a getitem node for any outputs that are unused.
+            # However, we need a name for them so that the number of outputs will
+            # correctly match the schema. Just assign a dummy name.
+            if idx not in idx_to_name:
+                idx_to_name[idx] = f"{node.name}_unused_{idx}"
+
+        arg_list = []
+        for i, element_meta_val in enumerate(meta_val):
+            arg_list.append(
+                self.export_tensor_output(idx_to_name[i], element_meta_val)
+            )
+
+        # Then, pack the return value differently depending on what the return type is.
+        if len(returns) == 1:
+            return_type = returns[0].real_type
+            assert isinstance(return_type, torch.ListType) and isinstance(
+                return_type.getElementType(), torch.TensorType
+            ), "Only tensors and lists of tensors supported"
+
+            return [Argument.create(as_tensors=arg_list)]
+        else:
+            assert all(
+                isinstance(ret.real_type, torch.TensorType) for ret in returns
+            ), f"Multiple returns can only have tensor returns, got: {[ret.real_type for ret in returns]}"
+
+            return [Argument.create(as_tensor=arg) for arg in arg_list]
+
+    def serialize(self, exported_program: ep.ExportedProgram) -> Tuple[GraphModule, bytes]:
+        for node in exported_program.graph.nodes:
+            try:
+                self.node = node
+                getattr(self, f"handle_{node.op}")(node)
+            except Exception as e:
+                if not isinstance(e, SerializeError):
+                    raise SerializeError(f"Failed serializing node {node}") from e
+
+        graph = Graph(
+            inputs=self.inputs,
+            nodes=self.nodes,
+            tensor_values=self.tensor_values,
+            sym_int_values=self.sym_int_values,
+            outputs=self.outputs,
+        )
+
+        return (
+            GraphModule(
+                graph=graph,
+                signature=export_signature(exported_program.graph_signature),
+                call_spec=export_call_spec(exported_program.call_spec),
+                opset_version=self.opset_version
+            ),
+            export_state_dict(exported_program.state_dict),
+        )
+
+
+def serialize(exported_program: ep.ExportedProgram) -> Tuple[GraphModule, bytes]:
```

Comment:
Q1: Should we pass in a version table here?

Q2: Shouldn't we only return a bytes object here?